File: python/src/sec_edgar_toolkit/types/current_events.py
"""Type definitions for 8-K current events parsing."""

from __future__ import annotations

from datetime import datetime
from typing import Any, Dict, List, Optional, TypedDict


class Event(TypedDict):
    """Current event information."""

    type: str
    description: str
    date: datetime
    item: str
    significance: str  # 'low', 'medium', 'high'
    details: Dict[str, Any]


class Agreement(TypedDict):
    """Material agreement information."""

    type: str
    parties: List[str]
    effective_date: datetime
    description: str
    value: Optional[float]
    currency: str


class PersonInfo(TypedDict):
    """Person information for executive changes."""

    name: str
    position: str
    previous_position: Optional[str]


class CompensationInfo(TypedDict):
    """Compensation information."""

    salary: float
    bonus: float
    equity: float


class ExecutiveChange(TypedDict):
    """Executive change information."""

    type: str  # 'appointment', 'resignation', 'termination'
    person: PersonInfo
    effective_date: datetime
    reason: Optional[str]
    compensation: Optional[CompensationInfo]


class AcquisitionTarget(TypedDict):
    """Acquisition target information."""

    name: str
    description: str


class Acquisition(TypedDict):
    """Acquisition information."""

    type: str  # 'acquisition', 'merger', 'divestiture'
    target: AcquisitionTarget
    value: Optional[float]
    currency: str
    expected_closing_date: Optional[datetime]
    status: str  # 'announced', 'pending', 'completed', 'terminated'


class GuidanceItem(TypedDict):
    """Earnings guidance item."""

    metric: str
    value: str


class EarningsData(TypedDict):
    """Earnings data information."""

    period: str
    revenue: Optional[float]
    net_income: Optional[float]
    earnings_per_share: Optional[float]
    guidance: List[GuidanceItem]


class ParsedCurrentEvent(TypedDict):
    """Complete parsed current event data."""

    form_type: str
    filing_date: datetime
    cik: str
    company_name: str
    ticker: str
    events: List[Event]
    material_agreements: List[Agreement]
    executive_changes: List[ExecutiveChange]
    acquisitions: List[Acquisition]
    earnings_results: Optional[EarningsData]

File: typescript/src/types/current-events.ts
/**
 * Type definitions for 8-K current events parsing
 */

export interface Event {
  type: string;
  description: string;
  date: Date;
  item: string;
  significance: 'low' | 'medium' | 'high';
  details: Record<string, any>;
}

export interface Agreement {
  type: string;
  parties: string[];
  effectiveDate: Date;
  description: string;
  value: number | null;
  currency: string;
}

export interface ExecutiveChange {
  type: 'appointment' | 'resignation' | 'termination';
  person: {
    name: string;
    position: string;
    previousPosition?: string;
  };
  effectiveDate: Date;
  reason?: string;
  compensation?: {
    salary: number;
    bonus: number;
    equity: number;
  };
}

export interface Acquisition {
  type: 'acquisition' | 'merger' | 'divestiture';
  target: {
    name: string;
    description: string;
  };
  value: number | null;
  currency: string;
  expectedClosingDate: Date | null;
  status: 'announced' | 'pending' | 'completed' | 'terminated';
}

export interface EarningsData {
  period: string;
  revenue: number | null;
  netIncome: number | null;
  earningsPerShare: number | null;
  guidance: {
    metric: string;
    value: string;
  }[];
}

export interface ParsedCurrentEvent {
  formType: string;
  filingDate: Date;
  cik: string;
  companyName: string;
  ticker: string;
  events: Event[];
  materialAgreements: Agreement[];
  executiveChanges: ExecutiveChange[];
  acquisitions: Acquisition[];
  earningsResults: EarningsData | null;
}